src/utils/user-agent.ts

```diff
@@ -19,5 +20,28 @@ export const buildUserAgent = () => {
     userAgentParts.push(`node ${process.version}`);
   }
 
+  if (config.sourceTag) {
+    userAgentParts.push(`source_tag=${normalizeSourceTag(config.sourceTag)}`);
+  }
+
   return userAgentParts.join('; ');
 };
+
+const normalizeSourceTag = (sourceTag: string) => {
```

Comment:
is this expected behavior from the customer setting the source tag?

Reply:
Yeah, it's mimicking what we're doing to the slug on other clients:
- https://github.com/pinecone-io/pinecone-java-client/blob/1480814d20be1b5359e806936ccb50c9de60281b/src/main/java/io/pinecone/configs/PineconeConfig.java#L92
- https://github.com/pinecone-io/pinecone-python-client/blob/58ea88cfcadc5914c137198b9a15c760b87edcc0/pinecone/utils/user_agent.py#L13

I'm not sure if this was officially capture but I'll send you the details I've got.